### ISSUE TITLE
Bug: failing to detect non-Latinic alphabets as not English

### ISSUE THREAD

**Issue by wadsaek** [Automattic/harper#254]
As far as i know, Harper checks whether a piece of text is likely to be English, and it fails when seeing Ukrainian.
I'm using harper 0.12.0 from nixpkgs

![image](https://github.com/user-attachments/assets/b55237c2-ee75-43f0-a7e5-31a7c73c7b83)
![image](https://github.com/user-attachments/assets/c2ef4d49-c1ca-4e9c-8544-e20270d995e4)

might be connected to #113

**Comment by wadsaek** [comment]
Same issue happens with Hebrew
![image](https://github.com/user-attachments/assets/74ad7687-acab-4c67-87f6-c25d1884f7b4)

**Comment by elijah-potter** [comment]
> As far as i know, Harper checks whether a piece of text is likely to be English, and it fails when seeing Ukrainian.

There is not yet a release of Harper that does this yet. The functionality exists in `master` but not in any official binary releases. Right now, I'm waiting on [Mason](https://github.com/mason-org/mason-registry/pull/7536) to pick up changes before pushing out anything new.

That said, once it _does_ get pushed out. I'll throw an update here.

**Comment by wadsaek** [comment]
can confirm there is a fix at harper 14 with the `isolateEnglish` option.

nevertheless, I'm still getting some weird behaviour

![Image](https://github.com/user-attachments/assets/042e6b02-22a8-4e0e-af55-1ab737400c7c)

harper is still trying to lint two of the words in the sentence

**Comment by elijah-potter** [comment]
That is quite odd. @wadsaek, would you mind giving me a copy-and-pasted version of that text? I'd like to add a test case before I start debugging.

**Comment by wadsaek** [comment]
> That is quite odd. @wadsaek, would you mind giving me a copy-and-pasted version of that text? I'd like to add a test case before I start debugging.

not right now, I won't have access to neovim for a while. however while I'm at it, i will probably test Hebrew and maybe some Arabic too to see if the issue will be reproducible with other languages

**Comment by elijah-potter** [comment]
Sounds good! Eager to hear back.

**Comment by wadsaek** [comment]
`Гарпере, ну, друг ти мій, не грайся ти зі мною так. Я ж тебе обов'язково знайду, не хвилюйся`
harper lints the words "Гарпере" and "ну". this is a different string, but the last one was a little worse in terms of Ukrainian, so i thought it'd be better to write a better piece of text

`אז מכאן זה די פשוט: או שאתה איתנו, או שאתה נגדנו. תחליט לבד, בלעדנו. יש לך איזה שעה וחצי בשביל לעשות את זה. שמעת?`
harper lints the words "בלעדנו" and "שמעת"

I will reach out if i get a sample of arabic that can be tested

I've noticed harper only does this to word ending in punctuation. I couldn't reproduce the issue without a punctuation marking. in particular, dots, commas, punctuation and question marks, as well as colons. semicolons don't trigger this, as well as other symbols and letters from other alphabets. not that it lints this every time. For example, `Гарпере, чому ти лінтиш лише слова із якимось там комами?` isn't linted

**Comment by elijah-potter** [comment]
I believe this was just fixed by #504. I'll run a release EOD today. Would you mind checking if the issue persists then?

**Comment by wadsaek** [comment]
yeah, of course. just ping me when you get a release out and I'll check it

**Comment by elijah-potter** [comment]
We just put a release out. If you use Mason, however, it could take a few hours for it to be available. Let me know when you have a chance to check! Thanks again.

**Comment by wadsaek** [comment]
I'm using nix and there hasn't been an update yet in nixpkgs, but i could easily build the release from source

**Comment by wadsaek** [comment]
none of the examples i provided before are linted at version 0.18.1
